pkgs/development/python-modules/json-stream-rs-tokenizer/default.nix

```
@@ -0,0 +1,40 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, rustPlatform
+, setuptools-rust
+}:
+
+buildPythonPackage rec {
+  pname = "json-stream-rs-tokenizer";
+  version = "0.4.13";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "smheidrich";
+    repo = "py-json-stream-rs-tokenizer";
+    rev = "v${version}";
+    hash = "sha256-9pJi80V7WKvsgtp0ffItWnjoOvFvfE/Sz6y2VlsU+wQ=";
+  };
+
+  postPatch = ''
+    cp ${./Cargo.lock} ./Cargo.lock
+  '';
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src postPatch;
+    name = "${pname}-${version}";
+    hash = "sha256-TjRdHSXHmF6fzCshX1I4Sq+A/fEmBHDPGZvJUxL13aM=";
+  };
+
+  nativeBuildInputs = [ setuptools-rust ] ++ (with rustPlatform; [ cargoSetupHook rust.cargo rust.rustc ]);
+
+  doCheck = false; # tests require json-stream -- technically we could fall back to its pure python impl to run these tests, but meh.
+
+  meta = with lib; {
+    description = "A faster tokenizer for the json-stream Python library";
+    homepage = "https://github.com/smheidrich/py-json-stream-rs-tokenizer";
+    license = licenses.mit;
+    maintainers = with maintainers; [ winter ];
+  };
+}
```

Infinite recursion escape.

```suggestion
let self = buildPythonPackage rec {
  pname = "json-stream-rs-tokenizer";
  version = "0.4.13";
  format = "setuptools";

  src = fetchFromGitHub {
    owner = "smheidrich";
    repo = "py-json-stream-rs-tokenizer";
    rev = "v${version}";
    hash = "sha256-9pJi80V7WKvsgtp0ffItWnjoOvFvfE/Sz6y2VlsU+wQ=";
  };

  postPatch = ''
    cp ${./Cargo.lock} ./Cargo.lock
  '';

  cargoDeps = rustPlatform.fetchCargoTarball {
    inherit src postPatch;
    name = "${pname}-${version}";
    hash = "sha256-TjRdHSXHmF6fzCshX1I4Sq+A/fEmBHDPGZvJUxL13aM=";
  };

  nativeBuildInputs = [ setuptools-rust ] ++ (with rustPlatform; [ cargoSetupHook rust.cargo rust.rustc ]);

  doCheck = false; # infinite recursion with json-stream
  
  passthru.tests = {
    doCheck = self.overridePythonAttrs (_: { doCheck = true; });
   };

  meta = with lib; {
    description = "A faster tokenizer for the json-stream Python library";
    homepage = "https://github.com/smheidrich/py-json-stream-rs-tokenizer";
    license = licenses.mit;
    maintainers = with maintainers; [ winter ];
  };
};
in self
```